src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -251,7 +258,7 @@ public async ValueTask DisposeAsync()
             }));
 
             _scope.Dispose();
-            Renderer.Dispose();
+            await Renderer.Invoke(() => Renderer.Dispose());
```

Comment:
This is something we could easily overlook (and evidently did overlook in the past). Is there now a test to ensure the disposal runs on the right sync context?

Reply:
I'll add one.

Reply:
Added one for this.